File: go-pr/48/go/code-quality-naming/string_processor.go
package main

import (
	"fmt"
	"strings"
)

type StringProcessor struct{}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (sp *StringProcessor) proc(str string) string {
	return strings.ToUpper(str)
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (sp *StringProcessor) ConvertToUpperCase(inputString string) string {
	return strings.ToUpper(inputString)
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (sp *StringProcessor) chk(s1, s2 string) bool {
	return strings.Contains(s1, s2)
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (sp *StringProcessor) ContainsSubstring(mainString, searchString string) bool {
	return strings.Contains(mainString, searchString)
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=1}
func (sp *StringProcessor) cnt(txt string, ch rune) int {
	c := 0
	for _, x := range txt {
		if x == ch {
			c++
		}
	}
	return c
}
// {/fact}

// {fact rule=code-quality-naming@v1.0 defects=0}
func (sp *StringProcessor) CountCharacterOccurrences(text string, character rune) int {
	count := 0
	for _, currentChar := range text {
		if currentChar == character {
			count++
		}
	}
	return count
}
// {/fact}

func main() {
	processor := &StringProcessor{}
	result := processor.proc("hello world")
	fmt.Println(result)
}

File: ts-pr/55/typescript/code-quality-naming/string_processor.ts
class StringProcessor {
    // {fact rule=code-quality-naming@v1.0 defects=1}
    proc(str: string): string {
        return str.toUpperCase();
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    convertToUpperCase(inputString: string): string {
        return inputString.toUpperCase();
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    chk(s1: string, s2: string): boolean {
        return s1.includes(s2);
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    containsSubstring(mainString: string, searchString: string): boolean {
        return mainString.includes(searchString);
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    cnt(txt: string, ch: string): number {
        let c = 0;
        for (let x of txt) {
            if (x === ch) {
                c++;
            }
        }
        return c;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    countCharacterOccurrences(text: string, character: string): number {
        let count = 0;
        for (let currentChar of text) {
            if (currentChar === character) {
                count++;
            }
        }
        return count;
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    spl(str: string, sep: string): string[] {
        return str.split(sep);
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=0}
    splitStringBySeparator(inputString: string, separator: string): string[] {
        return inputString.split(separator);
    }
    // {/fact}

    // {fact rule=code-quality-naming@v1.0 defects=1}
    trm(s: string): string {
        return s.trim();
    }
    // {/fact}
}

const processor = new StringProcessor();
const result = processor.proc('hello world');
console.log(result);